File: python/binary_search.py
from typing import List

def binarySearch(arr: List[int] , n: int) -> int:
    low = 0
    high = len(arr) -1

    while (low <= high):
        mid = (low + high) // 2
        if (arr[mid] == n):
            return mid
        if (n > arr[mid]):
            low = mid + 1
        if (n < arr[mid]):
            high = mid - 1

    return None


__all__ = [binarySearch]

File: typescript/binary-search.ts

function BinarySearch(arr: number[], n: number): number {
    let low = 0
    let high = arr.length

    while (low < high) {
        let mid = Math.floor((high + low) / 2)

        if (n == arr[mid]) {
            return mid
        }
        if (n > arr[mid]) {
            low = mid + 1
        }
        if (n < arr[mid]) {
            high = mid - 1
        }
    }

    return -1;
}

export default BinarySearch
